File: Project/Java/Power of God/src/power/of/god/BibPlans/Parser.java
/*
 * The MIT License
 *
 * Copyright 2016 apotter96.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package power.of.god.BibPlans;

import Books.Book;
import java.io.File;
import java.io.IOException;
import java.util.*;
import power.of.god.*;
import power.of.god.Settings.*;

/**
 *
 * @author apotter96
 */
public class Parser {
    public static ArrayList<String> PlanDays = new ArrayList<>();
    public static String HtmlText(String bibPlanFileName, int day) throws IOException
    {
        BibPlan bibPlan = BibPlanParser.BiblicalPlan(AppFiles.ReadAllText(new File(bibPlanFileName)));
        String html = "<html><head><title>(Bible Plan) " + bibPlan.Name + "</title></head>" +
                "<body>Today's Reading <b>";
        ArrayList<ArrayList<VerseObj>> vs = bibPlan.VerseList;
        VerseObj v1 = vs.get(day).get(0);
        VerseObj v2 = vs.get(day).get(1);
        html += v1.Book + " " + v1.Chapter + ":" + v1.Verse + " to " +
                v2.Book + " " + v2.Chapter + ":" + v2.Verse + " (" + Settings.BibleVersion + ")</b><br><br>" +
                Body(v1, v2);
        return html;
    }
    
    public static int GetBookIndex(String nameStr)
    {
        int loopCount = 0;
        for (Book b: Bible.Bible.AllBooks())
        {
            if (b.getName().equals(nameStr))
            {
                return loopCount;
            }
            loopCount++;
        }
        return -1; // Failed
    }
    
    public static Book GetBook(String n)
    {
        return Bible.Bible.AllBooks()[GetBookIndex(n)]; // Functional programming not needed
    }
    
    private static String Body(VerseObj v1, VerseObj v2)
    {
        return PatchClass.GetVerse(v1, v2);
    }
}

File: Windows/Beta-Abandoned/Power of God Lib/BibPlan/Parser.cs
using System.Collections.ObjectModel;
using System.IO;
using System.Linq;
using BibPlans;
using Power_of_God_Lib.NetBible;
using Power_of_God_Lib.NetBible.Books;
using Power_of_God_Lib.Utilities;

namespace Power_of_God_Lib.BibPlan
{
    public class Parser
    {
        /// <summary>
        /// The Bible Plan Collection used for counting current day.
        /// </summary>
        public static ObservableCollection<string> PlanDays = new ObservableCollection<string>();
        /// <summary>
        /// Returns the html of the Bible Plan 
        /// </summary>
        /// <param name="bibplanFileName">The file path and name</param>
        /// <param name="day">The day selected. Note that variable day is 0 based.</param>
        /// <returns></returns>
        public static string HtmlText(string bibplanFileName, int day)
        {
            var bibPlan = BibPlanParser.BiblicalPlan(File.ReadAllText(bibplanFileName));
            var html = "<html><head><title>(BiblePlan) " + bibPlan.Name + "</title></head>" +
                            "<body>Today's Reading: <b>";
            var vs = bibPlan.VerseList;
            var v1 = vs.ElementAt(day).ElementAt(0);
            var v2 = vs.ElementAt(day).ElementAt(1);
            html += v1.Book + " " + v1.Chapter + ":" + v1.Verse + " to " +
                v2.Book + " " + v2.Chapter + ":" + v2.Verse + "(" + Settings.UserSettings.scriptver + ")</b><br><br>" +
                Body(ToVerseObj(v1), ToVerseObj(v2));
             
            return html;
        }

        /// <summary>
        /// Converts between objects
        /// </summary>
        /// <param name="v"></param>
        /// <returns></returns>
        private static VerseObj ToVerseObj(VerseObj v)
        {
            return new VerseObj
            {
                Book = v.Book,
                Chapter = v.Chapter,
                Verse = v.Verse
            };
        }

        /// <summary>
        /// Get index of Bible book
        /// </summary>
        /// <param name="nameStr">The name of the bible book</param>
        /// <returns>The index</returns>
        private static int GetBookIndex(string nameStr)
        {
            var loopCount = 0;
            foreach (var b in Bible.AllBooks())
            {
                if (b.Name == nameStr)
                {
                    return loopCount;
                }
                loopCount += 1;
            }
            return -1;
        }
        /// <summary>
        /// Retrieve Bible Book for use with verse loading
        /// </summary>
        /// <param name="n">The name of the bible book</param>
        /// <returns>Bible Book</returns>
        private static Book GetBook(string n)
        {
            return Bible.AllBooks().FirstOrDefault(b => b.Name == n);
        }
        /// <summary>
        /// Gets the content between two verses
        /// </summary>
        /// <param name="v1">The starting verse</param>
        /// <param name="v2">The ending verse</param>
        /// <returns>The content between the two verses</returns>
        private static string Body(VerseObj v1, VerseObj v2)
        {
            
            return new PatchClass().GetVerse(
                new VerseObj
                {
                    Book = v1.Book,
                    Chapter = v1.Chapter,
                    Verse = v1.Verse
                },
                new VerseObj
                {
                    Book = v2.Book,
                    Chapter = v2.Chapter,
                    Verse = v2.Verse
                }
            );
        }
    }
}
